File: src/0121.best-time-to-buy-and-sell-stock.121/solution.ts
function maxProfit (prices: number[]): number {
    let prevMin = prices[0];
    let result = 0;
    for (let i = 1; i < prices.length; i++) {
        if (prices[i] > prevMin) {
            result = Math.max(result, prices[i] - prevMin);
        } else {
            prevMin = Math.min(prevMin, prices[i]);
        }
    }
    return result;
}

File: src/0121.best-time-to-buy-and-sell-stock.121/solution.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = prices[0];
        int result = 0;
        for (int i = 1; i < prices.length; i++) {
            result = Math.max(result, prices[i] - min);
            min = Math.min(min, prices[i]);
        }
        return result;
    }
}